sympy/geometry/ellipse.py

```diff
@@ -418,20 +475,47 @@ def equation(self, x='x', y='y'):
 
         arbitrary_point : Returns parameterized point on ellipse
 
+        References
+        ==========
+
+        .. [1] https://math.stackexchange.com/questions/108270/what-is-the-equation-of-an-ellipse-that-is-not-aligned-with-the-axis
+        .. [2] https://en.wikipedia.org/wiki/Ellipse#Equation_of_a_shifted_ellipse
+
         Examples
         ========
 
         >>> from sympy import Point, Ellipse
         >>> e1 = Ellipse(Point(1, 0), 3, 2)
         >>> e1.equation()
         y**2/4 + (x/3 - 1/3)**2 - 1
+        >>> e1.equation(slope = 1)
+        (-x + y + 1)**2/18 + (x + y - 1)**2/8 - 1
+        >>> e2 = Ellipse(Point(0, 0), 4, 1)
+        >>> e2.equation(slope = 1)
+        (-x + y)**2/32 + (x + y)**2/2 - 1
 
         """
-        x = _symbol(x, real=True)
-        y = _symbol(y, real=True)
-        t1 = ((x - self.center.x) / self.hradius)**2
-        t2 = ((y - self.center.y) / self.vradius)**2
-        return t1 + t2 - 1
+
+        if slope is not None:
+            a, b, x, y, x_c, y_c = symbols('a,b,x,y,x_c,y_c')
+            x_c = self.center.x
+            y_c = self.center.y
+
+            L = ((y - y_c) - slope * (x - x_c)) ** 2
+            l = (slope * (y - y_c) + (x - x_c)) ** 2
+
+            if idiff(L, y, x) == -1 / idiff(l, y, x):
```

Comment:
no need for if, just return the result AFAICT

Reply:
@smichr Done. Please take a look.